main.py

```diff
@@ -0,0 +1,361 @@
+
+
+@app.post("/token", tags=["User"])
+async def generate_token(request_form: OAuth2PasswordRequestForm = Depends()):
+    token = await token_generator(request_form.username, request_form.password)
+    return {"access_token": token, "token_type": "bearer"}
+
+
+async def get_current_user(token: str = Depends(oauth_scheme)):
+    return await very_token(token)
+
+
+@app.post("/users/me", tags=["User"])
+async def client_data(user: user_pydanticIn = Depends(get_current_user)):
+
+    business = await Business.get(owner=user)
+    logo = business.logo
+    logo = f'{SITE_URL}{logo}'
+
+    return {
+        "status": "ok",
+        "data": {
+            "username": user.username,
+            "email": user.email,
+            "is_verified": user.is_verified,
+            "join_date": user.join_date.strftime("%b %d %Y"),
+            "logo": logo,
+            "business": await business_pydantic.from_tortoise_orm(business)
+        }
+    }
+
+
+@app.get("/users/", tags=["User"], response_model=List[user_pydanticOut])
+async def get_users(user: user_pydanticIn = Depends(get_current_user),
+                    limit: int = Query(100, le=100),
+                    skip: int = Query(0, ge=0)
+                    ):
+
+    users = await User.filter(id__gt=skip, id__lte=skip+limit)
+    return users
+
+
+@post_save(User)
+async def create_business(
+        sender: "Type[User]",
+        instance: User,
+        created: bool,
+        using_db: "Optional[BaseDBAsyncClient]",
+        update_fields: List[str]) -> None:
+    if created:
+        business_obj = await Business.create(
+            business_name=instance.username,
+            owner=instance)
+        await business_pydantic.from_tortoise_orm(business_obj)
+        # send the email
+        await send_mail([instance.email], instance)
+
+
+@app.put("/business/{id}", response_model=business_pydantic, tags=["Business"])
+async def update_business(id: int,
+                          update_business: business_pydanticIn,
+                          user: user_pydantic = Depends(get_current_user)):
+
+    business = await Business.get(id=id)
+    owner = await business.owner
+
+    update_business = update_business.dict()
+
+    if user == owner:
+        await business.update_from_dict(update_business)
+        await business.save()
+        return await business_pydantic.from_tortoise_orm(business)
+
+    raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Not authenticated to perform this action",
+        headers={"WWW-Authenticate": "Bearer"}
+    )
+
+
+@app.post("/users/", tags=["User"], status_code=status.HTTP_201_CREATED, response_model=user_pydanticOut)
+async def user_registration(user: user_pydanticIn):
+    user_info = user.dict(exclude_unset=True)
+
+    # This is a bad way to do it:
+    # TODO fix this, create custom 'user_pydanticIn for validate data '
+    if len(user_info["password"]) < 8:
```

Comment:
**Security Issue**: The password length check is performed directly in the endpoint handler, which is a poor security practice as it mixes validation logic with business logic. <br> **Fix**: Implement a dedicated validation function or use Pydantic's validator to handle password length checks. This separates concerns and enhances security by centralizing validation logic. <br> **Code Suggestion**: 
 ```
 @@ -130,6 +130,7 @@
 # This is a bad way to do it:
 # TODO fix this, create custom 'user_pydanticIn for validate data '
 +@validator('password')
 +def validate_password(cls, value):
 +    if len(value) < 8:
 +        raise ValueError('Password must be at least 8 characters long')
 +    return value
 -if len(user_info["password"]) < 8:
 -    raise HTTPException(
 -        status_code=status.HTTP_400_BAD_REQUEST, detail="Password must be longer than 8 characters")
 ```